react/features/mobile/audio-mode/middleware.js

```diff
@@ -55,3 +59,32 @@ MiddlewareRegistry.register(({ getState }) => next => action => {
 
     return next(action);
 });
+
+/**
+ * Sets the audio mode after a failed conference based on the reason
+ * of the failure.
+ *
+ * @private
+ * @param {Object} state - The Redux state.
+ * @param {Object} action - The dispatched Redux action.
+ * @returns {Object | undefined}
+ */
+function _getFailedConferenceAudioMode(state: Object, action: Object) {
+    const { error } = action;
+
+    if (error.name === JitsiConferenceErrors.PASSWORD_REQUIRED) {
```

Comment:
this feature shouldn't be concerned about this. It should just read if the error is recoverable or not. In order to do that, you'll need to (carefully!) run next(action) **before**, so that field is set by the corresponding feature.

Reply:
What does "carefully" mean? Should I wrap it into a

carefully {
   // code comes here
}

block? ;)

Reply:
Ok, seriously, I think running next is dangerous here. My idea here is that the conference failed event should automagically know what errors are recoverable and it should be the feature's reducer's job to set that field. What do you think?

Reply:
why is running next dangerous here ? it's only about looking at the state after vs before the action is resolved. The thing is that the conference itself doesn't recover from those errors, but it can be recoverable only if specific features are turned on and are able to do the recovery. Those features are marking errors as recoverable from their middlewares.